Modules/_ctypes/malloc_closure.c

```diff
@@ -96,7 +97,11 @@ void Py_ffi_closure_free(void *p)
 {
 #ifdef HAVE_FFI_CLOSURE_ALLOC
 #ifdef USING_APPLE_OS_LIBFFI
+# ifdef HAVE_BUILTIN_AVAILABLE
     if (__builtin_available(macos 10.15, ios 13, watchos 6, tvos 13, *)) {
+#  else
+    if (&ffi_closure_free != NULL) {
```

Comment:
Is it ``&ffi_closure_free`` or ``ffi_closure_free`` which can be NULL?

Reply:
Testing for ffi_closure_free matches the pattern in other files, and in particular posixmodule.c where we have a longer history of weak linking (although the truly ancient parts of that were removed when adding support for arm64)